### DISCUSSION TITLE
How to load local NER model for visualisation using Streamlit?

### DISCUSSION THREAD

**Discussion by CarolLuca** [explosion/spaCy#8643] [Help: Coding & Implementations] [answered]
Hello!

I currently discovered [Streamlit](https://streamlit.io/) as a user-friendly and accessible way of sharing Machine Learning, Data Science, and Artificial Intelligence projects. So, I decided to deploy an app in order for me to have an easier process of sharing my SpaCy NER model's results and usage.

That is when I got stuck on a problem: I don't know how to load my model from a different device without having the actual files that construct it and form it. More exactly, how and where should I save my model so I can access my model with `spacy.load("that/path")`. 

I suppose it should be saved somewhere in cloud, but, not integrating a lot of cloud files into my projects, I don't consider myself secure enough to speculate. Also, this is the [repo](https://github.com/CarolLuca/StreamNERoLit.git) with the files for the streamlit app and I should mention that I am using SpaCy v3.0.1 (as I just saw it was released v3.1.0 - Congrats!).

I tried using GitHub, but some of the files were over the superior limit. I know there is a [way](https://git-lfs.github.com/) of pushing big files, but even if I succeeded in doing so, I wouldn't know how to call the model.

Any recommendation is welcomed!

**Comment by polm** [comment, marked as answer]
You'll basically want to copy your model to whatever machine you're running Streamlit on, but note that since Streamlit runs a webserver you can just access that server from other computers like any other website. So you can use some kind of cloud hosting, copy your model there, and run your server just like you would on your local dev machine. `spacy.load("that/path")` will work as usual since the files will be local on the server.

You might want to try asking about this on the Streamlit forum, since your question seems to be more about how to configure Streamlit.

**Reply by CarolLuca** [reply]
Thank you for the suggestion! I will readdress the question on the Streamlit forum.
